common/core/gpudevice.cpp

```diff
@@ -35,6 +35,8 @@ bool GpuDevice::Initialize() {
   if (initialization_state_ & kInitialized)
     return true;
 
+  lock_fd_ = open("/vendor/hwc.lock", O_RDONLY);
```

Comment:
Why not just do it in HandleRoutine as before ?

Reply:
You have locks here which should take care of synchronization

Reply:
Will add a lock for this. Existing code logic will make the thread exit at the end of Initialization method if lock_fd_ remains as default value -1. And HandleWait method relies on value of lock_fd_ to justify if it needs to fall into sleep. So initialize lock_fd_ early to ensure the rest of code logic works as expected.